dev-tools/xtask/src/a4x2_deploy.rs

```diff
@@ -0,0 +1,472 @@
+        .run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g0").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g1").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g2").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g3").run()?;
+
+    Ok(())
+}
+
+fn prepare_to_launch_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner 'prepare'").run()?;
+
+    // TODO could move these into the a4x2-package bundle.
+    // mostly of matter of having a consistent version of propolis.
+    exec_remote_script(&sh, "https://raw.githubusercontent.com/oxidecomputer/falcon/main/get-ovmf.sh")?;
+    exec_remote_script(&sh, "https://raw.githubusercontent.com/oxidecomputer/falcon/main/get-propolis.sh")?;
+
+    // Generate an ssh key we will use to log into the sleds.
+    cmd!(sh, "ssh-keygen -t ed25519 -N '' -f a4x2-ssh-key").run()?;
+
+    // Copy the public side into the cargo bay
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g0/root_authorized_keys"),
+    )?;
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g1/root_authorized_keys"),
+    )?;
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g2/root_authorized_keys"),
+    )?;
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g3/root_authorized_keys"),
+    )?;
+
+    Ok(())
+}
+
+fn try_launch_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    let _popdir = sh.push_dir(&env.a4x2_dir);
+
+    cmd!(sh, "pfexec ./a4x2 launch").run()?;
+
+    // XXX Is there a better way to do this?
+    let ce_addr_json =
+        cmd!(sh, "./a4x2 exec ce 'ip -4 -j addr show enp0s10'").read()?;
+
+    // Translated from this jq query:
+    // .[0].addr_info[] | select(.dynamic == true) | .local
+    let ce_addr_json: Value = serde_json::from_str(&ce_addr_json)?;
+    let customer_edge_addr = &ce_addr_json[0]["addr_info"]
+        .as_array()
+        .unwrap()
+        .iter()
+        .find(|v| v["dynamic"] == Value::Bool(true))
+        .ok_or(anyhow!("failed to find customer edge addr"))?["local"]
+        .as_str()
+        .unwrap();
+
+    // XXX Im told that pinging the gateway from inside the sleds is no longer
+    // necessary so I'm leaving it out for now. We'll see if that's true.
+    // So far it seems to be.
+
+    cmd!(sh, "pfexec route add 198.51.100.0/24 {customer_edge_addr}").run()?;
+
+    // Not sure how this IP is fixed, but it is
+    let api_url = "http://198.51.100.23";
+    println!("polling control plane for signs of life for up to 25 minutes");
+
+    // Print the date for the logs' benefit
+    let _ = cmd!(sh, "date").run();
+
+    // Timeout = (retries / 2) minutes
+    // XXX this is an arbitrary timeout. Should it be configurable? Skippable?
+    let mut retries = 50;
```

Comment:
Thinking about it, I feel like if the control plane doesn't come up we shouldn't tear down a4x2 (at least by default) -- instead, we should tell you that we aren't doing so, and then also tell you to run the stop command.